File: auto-pay-sdks/nodejs-sdk/src/models/QueryPaymentRequest.ts
/**
 * 查询支付请求
 */
export class QueryPaymentRequest {
  public readonly paymentId?: string;
  public readonly orderId?: string;
  public readonly status?: string;
  public readonly channel?: string;
  public readonly startTime?: string;
  public readonly endTime?: string;
  public readonly page?: number;
  public readonly size?: number;
  public readonly sort?: string;
  public readonly order?: 'asc' | 'desc';

  public constructor(builder: QueryPaymentRequestBuilder) {
    this.paymentId = builder.paymentId;
    this.orderId = builder.orderId;
    this.status = builder.status;
    this.channel = builder.channel;
    this.startTime = builder.startTime;
    this.endTime = builder.endTime;
    this.page = builder.page;
    this.size = builder.size;
    this.sort = builder.sort;
    this.order = builder.order;
  }

  /**
   * 创建构建器
   */
  public static builder(): QueryPaymentRequestBuilder {
    return new QueryPaymentRequestBuilder();
  }

  /**
   * 转换为查询参数
   */
  public toQueryParams(): Record<string, any> {
    const params: Record<string, any> = {};

    if (this.paymentId) params.payment_id = this.paymentId;
    if (this.orderId) params.order_id = this.orderId;
    if (this.status) params.status = this.status;
    if (this.channel) params.channel = this.channel;
    if (this.startTime) params.start_time = this.startTime;
    if (this.endTime) params.end_time = this.endTime;
    if (this.page) params.page = this.page;
    if (this.size) params.size = this.size;
    if (this.sort) params.sort = this.sort;
    if (this.order) params.order = this.order;

    return params;
  }
}

/**
 * 查询支付请求构建器
 */
export class QueryPaymentRequestBuilder {
  public paymentId?: string;
  public orderId?: string;
  public status?: string;
  public channel?: string;
  public startTime?: string;
  public endTime?: string;
  public page?: number;
  public size?: number;
  public sort?: string;
  public order?: 'asc' | 'desc';

  /**
   * 设置支付ID
   */
  public setPaymentId(paymentId: string): QueryPaymentRequestBuilder {
    this.paymentId = paymentId;
    return this;
  }

  /**
   * 设置订单ID
   */
  public setOrderId(orderId: string): QueryPaymentRequestBuilder {
    this.orderId = orderId;
    return this;
  }

  /**
   * 设置支付状态
   */
  public setStatus(status: string): QueryPaymentRequestBuilder {
    this.status = status;
    return this;
  }

  /**
   * 设置支付渠道
   */
  public setChannel(channel: string): QueryPaymentRequestBuilder {
    this.channel = channel;
    return this;
  }

  /**
   * 设置开始时间
   */
  public setStartTime(startTime: string): QueryPaymentRequestBuilder {
    this.startTime = startTime;
    return this;
  }

  /**
   * 设置结束时间
   */
  public setEndTime(endTime: string): QueryPaymentRequestBuilder {
    this.endTime = endTime;
    return this;
  }

  /**
   * 设置时间范围
   */
  public setTimeRange(startTime: string, endTime: string): QueryPaymentRequestBuilder {
    this.startTime = startTime;
    this.endTime = endTime;
    return this;
  }

  /**
   * 设置分页参数
   */
  public setPagination(page: number, size: number): QueryPaymentRequestBuilder {
    this.page = page;
    this.size = size;
    return this;
  }

  /**
   * 设置页码
   */
  public setPage(page: number): QueryPaymentRequestBuilder {
    this.page = page;
    return this;
  }

  /**
   * 设置每页大小
   */
  public setSize(size: number): QueryPaymentRequestBuilder {
    this.size = size;
    return this;
  }

  /**
   * 设置排序字段
   */
  public setSort(sort: string): QueryPaymentRequestBuilder {
    this.sort = sort;
    return this;
  }

  /**
   * 设置排序方向
   */
  public setOrder(order: 'asc' | 'desc'): QueryPaymentRequestBuilder {
    this.order = order;
    return this;
  }

  /**
   * 设置排序参数
   */
  public setSortBy(field: string, order: 'asc' | 'desc' = 'desc'): QueryPaymentRequestBuilder {
    this.sort = field;
    this.order = order;
    return this;
  }

  /**
   * 构建查询对象
   */
  public build(): QueryPaymentRequest {
    return new QueryPaymentRequest(this);
  }
}

/**
 * 退款请求
 */
export class RefundPaymentRequest {
  public readonly paymentId: string;
  public readonly amount?: number;
  public readonly reason?: string;
  public readonly metadata?: Record<string, any>;
  public readonly notifyUrl?: string;

  public constructor(builder: RefundPaymentRequestBuilder) {
    this.paymentId = builder.paymentId!;
    this.amount = builder.amount;
    this.reason = builder.reason;
    this.metadata = builder.metadata;
    this.notifyUrl = builder.notifyUrl;
  }

  /**
   * 创建构建器
   */
  public static builder(): RefundPaymentRequestBuilder {
    return new RefundPaymentRequestBuilder();
  }

  /**
   * 转换为JSON对象
   */
  public toJSON(): Record<string, any> {
    return {
      payment_id: this.paymentId,
      amount: this.amount,
      reason: this.reason,
      metadata: this.metadata,
      notify_url: this.notifyUrl
    };
  }
}

/**
 * 退款请求构建器
 */
export class RefundPaymentRequestBuilder {
  public paymentId?: string;
  public amount?: number;
  public reason?: string;
  public metadata?: Record<string, any>;
  public notifyUrl?: string;

  /**
   * 设置支付ID
   */
  public setPaymentId(paymentId: string): RefundPaymentRequestBuilder {
    this.paymentId = paymentId;
    return this;
  }

  /**
   * 设置退款金额
   */
  public setAmount(amount: number): RefundPaymentRequestBuilder {
    this.amount = amount;
    return this;
  }

  /**
   * 设置退款原因
   */
  public setReason(reason: string): RefundPaymentRequestBuilder {
    this.reason = reason;
    return this;
  }

  /**
   * 添加元数据
   */
  public setMetadata(metadata: Record<string, any>): RefundPaymentRequestBuilder {
    this.metadata = metadata;
    return this;
  }

  /**
   * 添加元数据项
   */
  public addMetadata(key: string, value: any): RefundPaymentRequestBuilder {
    if (!this.metadata) {
      this.metadata = {};
    }
    this.metadata[key] = value;
    return this;
  }

  /**
   * 设置通知URL
   */
  public setNotifyUrl(notifyUrl: string): RefundPaymentRequestBuilder {
    this.notifyUrl = notifyUrl;
    return this;
  }

  /**
   * 验证参数
   */
  private validate(): void {
    if (!this.paymentId || this.paymentId.trim() === '') {
      throw new Error('支付ID不能为空');
    }
  }

  /**
   * 构建退款对象
   */
  public build(): RefundPaymentRequest {
    this.validate();
    return new RefundPaymentRequest(this);
  }
}

File: auto-pay-sdks/java-sdk/src/main/java/com/autopay/sdk/model/request/QueryPaymentRequest.java
package com.autopay.sdk.model.request;

import com.fasterxml.jackson.annotation.JsonProperty;

/**
 * 查询支付请求模型
 */
public class QueryPaymentRequest {
    
    @JsonProperty("payment_id")
    private String paymentId;
    
    @JsonProperty("order_id")
    private String orderId;
    
    @JsonProperty("merchant_id")
    private String merchantId;
    
    @JsonProperty("channel")
    private String channel;
    
    @JsonProperty("status")
    private String status;
    
    @JsonProperty("start_time")
    private String startTime;
    
    @JsonProperty("end_time")
    private String endTime;
    
    @JsonProperty("page")
    private int page = 1;
    
    @JsonProperty("size")
    private int size = 20;
    
    @JsonProperty("sort_by")
    private String sortBy = "created_at";
    
    @JsonProperty("sort_order")
    private String sortOrder = "desc"; // asc, desc
    
    // 构造方法
    public QueryPaymentRequest() {}
    
    public QueryPaymentRequest(String paymentId) {
        this.paymentId = paymentId;
    }
    
    public QueryPaymentRequest(String paymentId, String orderId) {
        this.paymentId = paymentId;
        this.orderId = orderId;
    }
    
    // Builder模式
    public static Builder builder() {
        return new Builder();
    }
    
    public static class Builder {
        private final QueryPaymentRequest request = new QueryPaymentRequest();
        
        public Builder paymentId(String paymentId) {
            request.setPaymentId(paymentId);
            return this;
        }
        
        public Builder orderId(String orderId) {
            request.setOrderId(orderId);
            return this;
        }
        
        public Builder merchantId(String merchantId) {
            request.setMerchantId(merchantId);
            return this;
        }
        
        public Builder channel(String channel) {
            request.setChannel(channel);
            return this;
        }
        
        public Builder status(String status) {
            request.setStatus(status);
            return this;
        }
        
        public Builder startTime(String startTime) {
            request.setStartTime(startTime);
            return this;
        }
        
        public Builder endTime(String endTime) {
            request.setEndTime(endTime);
            return this;
        }
        
        public Builder page(int page) {
            request.setPage(page);
            return this;
        }
        
        public Builder size(int size) {
            request.setSize(size);
            return this;
        }
        
        public Builder sortBy(String sortBy) {
            request.setSortBy(sortBy);
            return this;
        }
        
        public Builder sortOrder(String sortOrder) {
            request.setSortOrder(sortOrder);
            return this;
        }
        
        public QueryPaymentRequest build() {
            return request;
        }
    }
    
    // Getter和Setter方法
    public String getPaymentId() {
        return paymentId;
    }
    
    public void setPaymentId(String paymentId) {
        this.paymentId = paymentId;
    }
    
    public String getOrderId() {
        return orderId;
    }
    
    public void setOrderId(String orderId) {
        this.orderId = orderId;
    }
    
    public String getMerchantId() {
        return merchantId;
    }
    
    public void setMerchantId(String merchantId) {
        this.merchantId = merchantId;
    }
    
    public String getChannel() {
        return channel;
    }
    
    public void setChannel(String channel) {
        this.channel = channel;
    }
    
    public String getStatus() {
        return status;
    }
    
    public void setStatus(String status) {
        this.status = status;
    }
    
    public String getStartTime() {
        return startTime;
    }
    
    public void setStartTime(String startTime) {
        this.startTime = startTime;
    }
    
    public String getEndTime() {
        return endTime;
    }
    
    public void setEndTime(String endTime) {
        this.endTime = endTime;
    }
    
    public int getPage() {
        return page;
    }
    
    public void setPage(int page) {
        this.page = page;
    }
    
    public int getSize() {
        return size;
    }
    
    public void setSize(int size) {
        this.size = size;
    }
    
    public String getSortBy() {
        return sortBy;
    }
    
    public void setSortBy(String sortBy) {
        this.sortBy = sortBy;
    }
    
    public String getSortOrder() {
        return sortOrder;
    }
    
    public void setSortOrder(String sortOrder) {
        this.sortOrder = sortOrder;
    }
    
    @Override
    public String toString() {
        return String.format("QueryPaymentRequest{paymentId='%s', orderId='%s', channel='%s', status='%s', page=%d, size=%d}", 
                paymentId, orderId, channel, status, page, size);
    }
}